conn.go

```diff
@@ -394,7 +399,6 @@ func (c *conn) close() {
 
 	// no errors
 	default:
```

Comment:
Just checking - after this change will Receiver.Receive() still return an error if I call it after a connection has been closed? Or is this just vestigial and that error gets propagated a different way?

Reply:
Correct (see test TestReceiverOnConnClosed).

The way this works is that when conn is closed (or faulted), `conn.Done` is closed.  The session detects this and sets `s.err = s.conn.Err()` which will be a `*ConnectionError`.  Then the session exits, and the error is propagated to its links in a similar way.